Add internet proxy configuration

tallandtree opened the pull request:
Internet proxy needs to be configured in systemd for Docker.
If inventory__environment.http_proxy is set, the proxy is configured in
/etc/systemd/system/docker.service.d/http-proxy.conf.
If inventory__environment.http_proxy is not set, the proxy configuration
is removed if set.

tallandtree committed:
Add internet proxy configuration

Internet proxy needs to be configured in systemd for Docker.
If inventory__environment.http_proxy is set, the proxy is configured in
/etc/systemd/system/docker.service.d/http-proxy.conf.
If inventory__environment.http_proxy is not set, the proxy configuration
is removed if set.

drybjed commented on tasks/main.yml:
```diff
@@ -83,6 +94,25 @@
+    group: 'root'
+    mode: '0644'
+  notify: [ 'Restart docker' ]
+  register: docker_register_systemd_service
```
I would use a separate variable name, for example `docker_register_systemd_proxy_present` to indicate changes to the file, and check all relevant variables in the `systemctl daemon-reload` task. Using the same variable names might result in wrong configuration in certain situations.

drybjed commented on tasks/main.yml:
```diff
@@ -83,6 +94,25 @@
+  file:  path='/etc/systemd/system/docker.service.d/http-proxy.conf' state=absent
+  when: (docker_upstream|d() | bool) and 
+    (inventory__environment is not defined or inventory__environment.http_proxy is not defined)
+  tags: [ 'role::docker:config' ]
```
This task also needs to be checked for changes to detect if `systemd` needs to be reloaded.

drybjed commented:
@tallandtree Very nice work. :+1:

tallandtree committed:
Add separate registration for proxy configuration

Added a separate register eventname for configuring and removing proxy.
Also added "Restart docker" notification in 'Reload systemctl
daemon-reload' as docker needs to be restarted after the daemon is
reloaded.

drybjed commented on tasks/main.yml:
```diff
@@ -71,24 +71,58 @@
+         (docker_register_systemd_proxy_present|d() and
+         docker_register_systemd_proxy_present.changed) or
+         (docker_register_systemd_proxy_absent|d() and
+         docker_register_systemd_proxy_absent.changed)))
```
These can be written as `docker_register_systemd_proxy_absent|changed`, this will also ensure correct variable type resolution by Ansible.

tallandtree committed:
Ensure correct variable type resolution by Ansible

Changed .changed to |changed for proxy configuration parts.
